salt/orchestrate/pnda.sls

```diff
@@ -285,3 +285,19 @@ orchestrate-pnda-install_remove_new_node_markers:
     - sls: orchestrate.remove_new_node_marker
     - timeout: 120
     - queue: True
+
+orchestrate-pnda-oozie-ssh_keygen:
+  salt.state:
+    - tgt: 'G@pnda_cluster:{{pnda_cluster}} and G@roles:oozie_database'
```

Comment:
As of now, there is no role as "oozie" or "oozie_server" and in pico and std oozie_database and oozie server are running on the same machine. So we decided to use the existing oozie_database role.

Reply:
oozie placement is controlled by the cluster manager (cloudera manager or ambari) you could look at using get_hosts_by_hadoop_role helper function. The oozie_database role isn't the right one to use as it would only be working by coincidence. It's a bit worrying that in the absence of something sensible you decided to do something random that sounds sort of right.

Reply:
pnda.get_hosts_by_hadoop_role is a custom module function, we can't use the custom module function in salt.orchestrate because it is running in server mode. So we will plan to add a new role as "oozie_server" to respective nodes and when we install oozie's server and database to different nodes, we can modify this role to respective nodes.

Reply:
I agree that is an annoying limitation we've hit before, if another role is the only way then I guess, but that means it needs to be kept in sync with the blueprint config which is also annoying.